File: abc117/d/d.cs
using System;
class Program
{
	static void Main(string[] args)
	{
		string[] firstLine = Console.ReadLine().Split(' ');
		string[] secondLine = Console.ReadLine().Split(' ');
        long N = long.Parse(firstLine[0]);
        long K = long.Parse(firstLine[1]);

        long[] A = new long[N];

        for (long i = 0; i < N; i++) {
            A[i] = long.Parse(secondLine[i]);
        }

        long ans = 0;

        for (long x = 0; x <= K; x++) {
            long fx = 0;
            for (long i = 0; i < N; i++) {
                fx += x ^ A[i];
            }

            if (fx > ans) { ans = fx; }
        }

		//出力
		Console.WriteLine(ans);
	}
}

File: abc117/d/d.js
"use strict";

// read lines from stdin
const input = require("fs")
  .readFileSync("/dev/stdin", "utf8")
  .split("\n");

const firstLine = input[0].split(" ").map((x) => parseInt(x, 10));

const N = firstLine[0];
const K = firstLine[1];
const A = input[1].split(" ").map((x) => parseInt(x, 10));

let ans = A.reduce((pre, cur) => pre + cur);

for (let x = 1; x <= K; x++) {
  let fx = 0;
  for (let i = 0; i < N; i++) {
    fx += x ^ A[i];
  }

  if (fx > ans) { ans = fx; }
}

console.log(ans);
